### DISCUSSION TITLE
How to release resources from the ViewModel

### DISCUSSION THREAD

**Discussion by songminkyu** [cefsharp/CefSharp#3442] [General]
Hi. Everyone
We are implementing the web browser and PDF preview function using the CefSharp MVVM method. However, after loading the PDF, I want to release resources depending on whether they are needed, but I want to know what the "ViewModel" handles.
As a temporary alternative, there is currently no method, so it is being used by calling dispose in the code behind it. Any help would be appreciated. :)

**Comment by amaitland** [comment]
> , I want to release resources depending on whether they are needed, 

What resources?  Please clarify. 

> I want to release resources depending on whether they are needed, but I want to know what the "ViewModel" handles.

What do you mean by this?

**Reply by amaitland** [reply]
Generally speaking to ensure the browser is disposed correctly I suggest binding the http://cefsharp.github.io/api/88.2.x/html/P_CefSharp_Wpf_ChromiumWebBrowser_CleanupElement.htm property to an element that has the same lifespan,  e.g. a tab that's closed.

The viewmodel shouldn't be involved directly when disposing the browser.

**Reply by songminkyu** [reply]
If you call FrameworkElement, can you see the dispose effect?

**Reply by amaitland** [reply]
> If you call FrameworkElement, can you see the dispose effect?

I don't understand the question.
